spec/datadog_spec.rb

```diff
@@ -66,16 +66,17 @@
       it 'posts an event' do
```

Comment:
In this file you can probably generalize tests with `before` and `context`. 

Why are we mocking the dogapi gem internal behavior? Shouldn't we use public gem API and mock it only?